### ISSUE TITLE
ByteOrderedPartitioner

### ISSUE THREAD

**Issue by lucilecoutouly** [OpenTSDB/asynccassandra#6]
Hey,

when y try to run opentsdb with cassandra with ByteOrderedPartitioner it fails
without ByteOrderedPartitioner it succeed 

what's wrong ?

```
:/usr/share/opentsdb# /usr/share/opentsdb/build/tsdb tsd --conf
ig=/etc/opentsdb/opentsdb.conf --port=4242 --staticroot=static/ --cachedir=/tmp
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/share/opentsdb/build/third_party/logback/
logback-classic-1.0.13.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/share/opentsdb/build/third_party/asynccas
sandra/asynccassandra-0.0.1-20160229.001338-4-jar-with-dependencies.jar!/org/slf
4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorSta
ticBinder]
2017-01-23 10:21:55,901 INFO  [main] TSDMain: Starting.
2017-01-23 10:21:55,904 INFO  [main] TSDMain: net.opentsdb.tools 2.3.0 built at
revision cac608a (MINT)
2017-01-23 10:21:55,904 INFO  [main] TSDMain: Built on 2017/01/20 14:16:41 +0000
 by root@33842653d6db:/usr/share/opentsdb/build
2017-01-23 10:21:55,906 INFO  [main] Config: Successfully loaded configuration f
ile: /etc/opentsdb/opentsdb.conf
2017-01-23 10:21:55,969 INFO  [main] Config: Successfully loaded configuration f
ile: /etc/opentsdb/opentsdb.conf
2017-01-23 10:21:56,031 WARN  [main] PluginLoader: Unable to locate any plugins
of the type: net.opentsdb.query.filter.TagVFilter
2017-01-23 10:21:56,096 INFO  [main] ConnectionPoolMBeanManager: Registering mbe
an: com.netflix.MonitoredResources:type=ASTYANAX,name=MyConnectionPool,ServiceTy
pe=connectionpool
2017-01-23 10:21:56,109 INFO  [main] CountingConnectionPoolMonitor: AddHost: cas
sandra
2017-01-23 10:21:56,179 INFO  [main] CountingConnectionPoolMonitor: AddHost: 10.
42.141.144
2017-01-23 10:21:56,179 INFO  [main] CountingConnectionPoolMonitor: RemoveHost:
cassandra
2017-01-23 10:21:56,195 INFO  [main] CountingConnectionPoolMonitor: AddHost: cas
sandra
2017-01-23 10:21:56,211 INFO  [main] CountingConnectionPoolMonitor: AddHost: 10.
42.141.144
2017-01-23 10:21:56,211 INFO  [main] CountingConnectionPoolMonitor: RemoveHost:
cassandra
2017-01-23 10:21:56,230 INFO  [main] TSDB: Flushing compaction queue
2017-01-23 10:21:56,235 ERROR [main] ConnectionPoolMBeanManager: com.netflix.Mon
itoredResources:type=ASTYANAX,name=MyConnectionPool,ServiceType=connectionpool
2017-01-23 10:21:56,236 INFO  [main] TSDB: Completed shutting down the TSDB
Exception in thread "main" java.lang.RuntimeException: Initialization failed
        at net.opentsdb.tools.TSDMain.main(TSDMain.java:237)
Caused by: com.stumbleupon.async.DeferredGroupException: At least one of the Def
erreds failed, first exception:
        at com.stumbleupon.async.DeferredGroup.done(DeferredGroup.java:169)
        at com.stumbleupon.async.DeferredGroup.recordCompletion(DeferredGroup.ja
va:142)
        at com.stumbleupon.async.DeferredGroup.access$000(DeferredGroup.java:36)
        at com.stumbleupon.async.DeferredGroup$1Notify.call(DeferredGroup.java:8
2)
        at com.stumbleupon.async.Deferred.doCall(Deferred.java:1278)
        at com.stumbleupon.async.Deferred.addCallbacks(Deferred.java:688)
        at com.stumbleupon.async.Deferred.addBoth(Deferred.java:772)
        at com.stumbleupon.async.DeferredGroup.<init>(DeferredGroup.java:119)
        at com.stumbleupon.async.Deferred.group(Deferred.java:897)
        at net.opentsdb.core.TSDB.checkNecessaryTablesExist(TSDB.java:593)
        at net.opentsdb.tools.TSDMain.main(TSDMain.java:195)
Caused by: java.util.concurrent.ExecutionException: com.netflix.astyanax.connect
ionpool.exceptions.NoAvailableHostsException: NoAvailableHostsException: [host=N
one(0.0.0.0):0, latency=0(0), attempts=0]No hosts to borrow from
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at org.hbase.async.HBaseClient$1ResponseCB.run(HBaseClient.java:170)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.
java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor
.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: com.netflix.astyanax.connectionpool.exceptions.NoAvailableHostsExcept
ion: NoAvailableHostsException: [host=None(0.0.0.0):0, latency=0(0), attempts=0]
No hosts to borrow from
        at com.netflix.astyanax.connectionpool.impl.RoundRobinExecuteWithFailove
r.<init>(RoundRobinExecuteWithFailover.java:48)
        at com.netflix.astyanax.connectionpool.impl.RoundRobinConnectionPoolImpl
.newExecuteWithFailover(RoundRobinConnectionPoolImpl.java:66)
        at com.netflix.astyanax.connectionpool.impl.AbstractHostPartitionConnect
ionPool.executeWithFailover(AbstractHostPartitionConnectionPool.java:352)
        at com.netflix.astyanax.thrift.ThriftColumnFamilyQueryImpl$1.execute(Thr
iftColumnFamilyQueryImpl.java:186)
        at com.netflix.astyanax.thrift.ThriftColumnFamilyQueryImpl$1$4.call(Thri
ftColumnFamilyQueryImpl.java:297)
        at com.netflix.astyanax.thrift.ThriftColumnFamilyQueryImpl$1$4.call(Thri
ftColumnFamilyQueryImpl.java:294)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        ... 3 more
```

Thanks

**Comment by manolama** [comment]
Which version of Cassandra are you running? It could be the version you have removed that partitioner.

**Comment by lucilecoutouly** [comment]
i try with different version from 1.7 to 2.2 
i succeed creating database cassandra with different partitionner
but
i can't run opentsdb over the byte partitionner
i can run opentsdb over murmur partitioner

i don't understand why ?

**Comment by jpdurot** [comment]
I tried with Cassandra 3.10 with success. Did you change partitioner parameter before starting Cassandra for the first time ? I know when I changed this setting after running Cassandra with another partitioner first, I got an error message (but don't remember the message, sorry)
